Sources/PayPalNativePayments/PayPalNativeCheckoutClient.swift

```diff
@@ -66,18 +66,28 @@ public class PayPalNativeCheckoutClient {
                     self.notifySuccess(for: result)
                 },
                 onShippingChange: { shippingChange, shippingChangeAction in
-                    shippingChangeAction.approve()
-                    
                     switch shippingChange.type {
                     case .shippingAddress:
+                        shippingChangeAction.approve()
                         let shippingAddress = PayPalNativeShippingAddress(shippingChange.selectedShippingAddress)
                         self.notifyShippingChange(shippingAddress: shippingAddress)
                         
                     case .shippingMethod:
-                        if let selectedShippingMethod = shippingChange.selectedShippingMethod {
-                            let shippingMethod = PayPalNativeShippingMethod(selectedShippingMethod)
-                            self.notifyShippingMethod(shippingMethod: shippingMethod)
+                        guard let selectedShippingMethod = shippingChange.selectedShippingMethod else {
+                            return
                         }
+                        
```

Comment:
we need to do a call to orders api, using the orderID provided in `PayPalNativeCheckoutRequest`. Once you have the order, we need to get the `amount` and add it with the shipping method cost, and add it to the patch request. Something like this:
```swift
    let orderAmount = // get order amount by calling orders v2 using orderID
    let selectedShippingOptionCost = Double(shippingChange.selectedShippingMethod?.amount?.value ?? "0") ?? 0
    let newTotal = orderAmount + selectedShippingOptionCost
    let patchRequest = PatchRequest()
    patchRequest.replace(amount: PurchaseUnit.Amount(currencyCode: usd, value: newTotal))
```

Reply:
Gotcha - looks like NXO currently offers an [order amount `patch`](https://github.paypal.com/nativexo/ios-native-checkout/blob/effc3b12cfd2ac6382b48ffa6fd1f23e63007ca8/Sources/PayPalCheckout/Sources/OrderActions/PatchAction/PatchRequest.swift#L114-L128) too we could leverage (for now)